src/datomic_rest_api/rest/fields/gene.clj

```diff
@@ -2085,81 +2087,128 @@
+                          expr-pattern (->>
+                                         (q '[:find [?e ...]
+                                              :in $ ?f
+                                              :where [?ef :expr-pattern.associated-feature/feature ?f]
+                                              [?e :expr-pattern/associated-feature ?ef]
+                                              [?e :expr-pattern/anatomy-term _]]
+                                            db fid)
+                                         (map
+                                           (fn [eid]
+                                             (let [expr (entity db eid)]
+                                               {:text (map #(pack-obj "anatomy-term" (:expr-pattern.anatomy-term/anatomy-term %))
+                                                           (:expr-pattern/anatomy-term expr))
+                                                :evidence {:by (pack-obj "expr-pattern" expr)}})))
+                                         (seq))
+                          bounded-by (->> (:feature/bound-by-product-of feature)
+                                          (map #(pack-obj "gene" (:feature.bound-by-product-of/gene %)))
+                                          (seq))
+                          tf (if (contains? feature :feature/associated-with-transcription-factor)
+                               (pack-obj "transcription-factor" (:feature.associated-with-transcription-factor/transcription-factor
+                                                                  (first (:feature/associated-with-transcription-factor feature))))
+                               nil)]
+                      {:name (pack-obj "feature" feature)
+                       :description (first (:feature/description feature))
+                       :method (if (nil? method) nil (str/replace method #"_" " "))
+                       :interaction (if (nil? interaction) [] interaction)
+                       :expr_pattern (if (nil? expr-pattern) [] expr-pattern)
+                       :bounded_by (if (nil? bounded-by) [] bounded-by)
+                       :tf tf })))
+                (seq))]
+    {:data (if (nil? data) nil data)
      :description
      "Features associated with this Gene"}))
 
+(defn- get-segments [gene]
+  (let [g-species (helpers.species/parse-species-name (:species/id (:gene/species gene)))
+        sequence-database (datomic-rest-api.db.sequence/get-default-sequence-database g-species)
+        features (datomic-rest-api.helpers.sequence/sequence-features sequence-database (:gene/id gene))]
+    features))
+
+(defn- longest-segment [segments]
+  (first
+      (sort-by #(- (:start %) (:end %)) segments)))
+
+(defn- get-segment [gene]
+  (let [segments (get-segments gene)
+        segment (longest-segment segments)]
+    segment))
+
+(defn- segment-to-position [gene segment gbrowse]
+  (let [start (if (< (:start segment) (:end segment))
```

Comment:
Find the assignment of `start` and `stop` here confusing (not sure why `:start` can sometime be < `:end` in `segment`?
Anyhow, consider using:
```clojure
[[start, stop] (->> segment ((juxt :start :stop)) (sort-by -)) 
```